### ISSUE TITLE
Build fails with :EOF: Error: Exponent does not have any digits

### ISSUE THREAD

**Issue by tawe141** [CGAL/cgal-swig-bindings#133]
Configuration: 

- Mac OSX 10.14
- CMake 3.15.4
- clang-1001.0.46.4
- Python 3.7 (Miniconda3)

`cmake` works great, but `make` fails during compilation. Here is the cmake command I used:

```
cmake -DCGAL_DIR=/usr/local/lib -DBUILD_PYTHON=ON -DBUILD_JAVA=OFF -DCMAKE_BUILD_TYPE=Release ../..
```

and here is the output of `make`:

```
Scanning dependencies of target CGAL_Kernel_cpp
[  1%] Building CXX object SWIG_CGAL/Kernel/CMakeFiles/CGAL_Kernel_cpp.dir/Point_2.cpp.o
[  2%] Building CXX object SWIG_CGAL/Kernel/CMakeFiles/CGAL_Kernel_cpp.dir/Weighted_point_2.cpp.o
[  3%] Building CXX object SWIG_CGAL/Kernel/CMakeFiles/CGAL_Kernel_cpp.dir/Segment_2.cpp.o
[  4%] Building CXX object SWIG_CGAL/Kernel/CMakeFiles/CGAL_Kernel_cpp.dir/Triangle_2.cpp.o
[  5%] Building CXX object SWIG_CGAL/Kernel/CMakeFiles/CGAL_Kernel_cpp.dir/Ray_2.cpp.o
[  6%] Building CXX object SWIG_CGAL/Kernel/CMakeFiles/CGAL_Kernel_cpp.dir/Direction_2.cpp.o
[  7%] Building CXX object SWIG_CGAL/Kernel/CMakeFiles/CGAL_Kernel_cpp.dir/Line_2.cpp.o
[  8%] Building CXX object SWIG_CGAL/Kernel/CMakeFiles/CGAL_Kernel_cpp.dir/Vector_2.cpp.o
[  9%] Building CXX object SWIG_CGAL/Kernel/CMakeFiles/CGAL_Kernel_cpp.dir/Polygon_2.cpp.o
[ 10%] Building CXX object SWIG_CGAL/Kernel/CMakeFiles/CGAL_Kernel_cpp.dir/Bbox_2.cpp.o
[ 11%] Building CXX object SWIG_CGAL/Kernel/CMakeFiles/CGAL_Kernel_cpp.dir/Point_3.cpp.o
[ 12%] Building CXX object SWIG_CGAL/Kernel/CMakeFiles/CGAL_Kernel_cpp.dir/Weighted_point_3.cpp.o
[ 13%] Building CXX object SWIG_CGAL/Kernel/CMakeFiles/CGAL_Kernel_cpp.dir/Sphere_3.cpp.o
[ 14%] Building CXX object SWIG_CGAL/Kernel/CMakeFiles/CGAL_Kernel_cpp.dir/Plane_3.cpp.o
[ 15%] Building CXX object SWIG_CGAL/Kernel/CMakeFiles/CGAL_Kernel_cpp.dir/Segment_3.cpp.o
[ 17%] Building CXX object SWIG_CGAL/Kernel/CMakeFiles/CGAL_Kernel_cpp.dir/Line_3.cpp.o
[ 18%] Building CXX object SWIG_CGAL/Kernel/CMakeFiles/CGAL_Kernel_cpp.dir/Triangle_3.cpp.o
[ 19%] Building CXX object SWIG_CGAL/Kernel/CMakeFiles/CGAL_Kernel_cpp.dir/Tetrahedron_3.cpp.o
[ 20%] Building CXX object SWIG_CGAL/Kernel/CMakeFiles/CGAL_Kernel_cpp.dir/Direction_3.cpp.o
[ 21%] Building CXX object SWIG_CGAL/Kernel/CMakeFiles/CGAL_Kernel_cpp.dir/Ray_3.cpp.o
[ 22%] Building CXX object SWIG_CGAL/Kernel/CMakeFiles/CGAL_Kernel_cpp.dir/Vector_3.cpp.o
[ 23%] Building CXX object SWIG_CGAL/Kernel/CMakeFiles/CGAL_Kernel_cpp.dir/Bbox_3.cpp.o
[ 24%] Building CXX object SWIG_CGAL/Kernel/CMakeFiles/CGAL_Kernel_cpp.dir/Object.cpp.o
[ 25%] Building CXX object SWIG_CGAL/Kernel/CMakeFiles/CGAL_Kernel_cpp.dir/global_functions.cpp.o
[ 26%] Building CXX object SWIG_CGAL/Kernel/CMakeFiles/CGAL_Kernel_cpp.dir/Origin.cpp.o
[ 27%] Building CXX object SWIG_CGAL/Kernel/CMakeFiles/CGAL_Kernel_cpp.dir/Iso_rectangle_2.cpp.o
[ 28%] Linking CXX shared library ../../lib/libCGAL_Kernel_cpp.dylib
[ 28%] Built target CGAL_Kernel_cpp
Scanning dependencies of target CGAL_Kernel_python_swig_compilation
[ 29%] Swig compile CGAL_Kernel.i for python
[ 29%] Built target CGAL_Kernel_python_swig_compilation
Scanning dependencies of target CGAL_Kernel_python
[ 30%] Building CXX object SWIG_CGAL/Kernel/CMakeFiles/CGAL_Kernel_python.dir/__/__/build-python/CGAL/CGAL_KernelPYTHON_wrap.cxx.o
[ 31%] Linking CXX shared module ../../build-python/CGAL/_CGAL_Kernel.so
[ 31%] Built target CGAL_Kernel_python
Scanning dependencies of target CGAL_Triangulation_3_cpp
[ 32%] Building CXX object SWIG_CGAL/Triangulation_3/CMakeFiles/CGAL_Triangulation_3_cpp.dir/Object.cpp.o
[ 34%] Linking CXX shared library ../../lib/libCGAL_Triangulation_3_cpp.dylib
[ 34%] Built target CGAL_Triangulation_3_cpp
Scanning dependencies of target CGAL_Triangulation_3_python_swig_compilation
[ 35%] Swig compile CGAL_Triangulation_3.i for python
:EOF: Error: Exponent does not have any digits
make[2]: *** [SWIG_CGAL/Triangulation_3/CMakeFiles/CGAL_Triangulation_3_python.dir/CGAL_Triangulation_3PYTHON.stamp] Error 1
make[2]: *** Deleting file `SWIG_CGAL/Triangulation_3/CMakeFiles/CGAL_Triangulation_3_python.dir/CGAL_Triangulation_3PYTHON.stamp'
make[1]: *** [SWIG_CGAL/Triangulation_3/CMakeFiles/CGAL_Triangulation_3_python_swig_compilation.dir/all] Error 2
make: *** [all] Error 2
```

I've tried looking up the last error which appears to be from `swig`, but no results seem relevant. How do I fix this error?

**Comment by sloriot** [comment]
Could it be that you run out of disk space? What SWIG version are you using?

**Comment by tawe141** [comment]
No, I should have plenty of hard drive space. SWIG version is 4.0.1.

**Comment by lrineau** [comment]
@maxGimeno Can you please update the CI of cgal-swig-bindings for macOS? The concurrent configuration, that I made, uses Python provided by macOS, but that is a Python 2.7, and macOS will soon stop shipping any Python version is its system.

In the CI, install Python 3.7 using brew. There is a "homebrew addon" package in Travis that can be used. Than we'll see if we can reproduce the error.

**Comment by xantares** [comment]
its the cgal bad version.h issue, see https://github.com/CGAL/cgal/issues/4355

**Comment by sloriot** [comment]
@xantares @tawe141 could you please try #139 and let me know if it fixes the issue (I cannot reproduce it). Thanks.

**Comment by xantares** [comment]
nope, as I said the problem is with CGAL_GIT_HASH

**Comment by lrineau** [comment]
This bug has been fixed directly in CGAL, by CGAL/cgal#4475 (merged in CGAL-5.0.1), and CGAL/cgal@17a58a5625178b11e65aa96b6c0cfea3450ac2e6 (for 4.14.3). So you should upgrade CGAL to version 4.14.3 or 5.0.1, or later versions.
